code/modules/projectiles/gun.dm

```diff
@@ -568,8 +582,131 @@ As sniper rifles have both and weapon mods can change them as well. ..() deals w
 	update_mag_overlay()
 	update_attachables()
 
+//-- code by carlarc, slightly modified, emphasis on slightly --//
+
+// function to update durability flag, blatant shit code
+/obj/item/weapon/gun/proc/update_gun_durability()
+	gun_durability = clamp(gun_durability, GUN_DURABILITY_BROKEN, GUN_DURABILITY_MAX)
+
+// call update_gun_durability() whenever gun_durability changes
+/obj/item/weapon/gun/proc/set_gun_durability(new_gun_durability)
+	gun_durability = new_gun_durability
+	update_gun_durability()
+
+/obj/item/weapon/gun/proc/check_jam(mob/living/user)
+	if(gun_durability <= GUN_DURABILITY_BROKEN) //prevents firing without spamming your screen with both jamming and worn out noises
+		check_worn_out(user)
+		return NONE
+	if(jammed)
+		if(world.time % 3)
+			playsound(src, 'sound/weapons/handling/gun_jam_click.ogg', 35, TRUE)
+			to_chat(user, SPAN_WARNING("Your gun is jammed! Mash Unique-Action to unjam it!"))
```

Comment:
Would be better to  dynamically replace with the user's keybind for unique-action; e.g. Mash [unique_action_key] where unique_action_key is set to the user's keybind--

So if it was set to the default(spacebar) it'd say "Your gun is jammed! Mash space to unjam it!"